src/hotspot/os/linux/globals_linux.hpp

```diff
@@ -77,7 +77,18 @@
           "Use CPU_ALLOC code path in os::active_processor_count ")     \
                                                                         \
   product(bool, DumpPerfMapAtExit, false, DIAGNOSTIC,                   \
-          "Write map file for Linux perf tool at exit")
+          "Write map file for Linux perf tool at exit")                 \
+                                                                        \
+  product(intx, TimerSlack, -1, EXPERIMENTAL,                           \
+          "(Linux-specific) Override the timer slack value to this "    \
+          "number of nanoseconds. Lower value provides more accurate "  \
+          "high-precision timers, at the expense of (possibly) worse "  \
+          "power efficiency. Set -1 to disable the override. In current "\
+          "Linux, 0 means using the system-wide default, which is the " \
+          "same as disabled override. See prctl(PR_SET_TIMERSLACK) for "\
+          "more info.")                                                 \
+                                                                        \
```

Comment:
None of the other Linux specifc args write "(Linux specific)". I'd leave that out.

The first sentence is difficult to parse. How about "Overrides timer slack value to the given number of nanoseconds [1..n]."

Why do we need two values (-1, 0) for oob? Is the "In current Linux, 0..." info useful for anything? I'd just use "0" as "default".

Reply:
IIUC using zero as default would be incorrect as that may reset an inherited customized value to the platform default. A value of -1 says "ignore this".

Reply:
We only call pctrl if the flag is >0. So, 0 has already the meaning of "ignore".

Reply:
That seems a bug then. We should pass zero to pctrl if that is the value set (even if the current implementation does nothing for a value of zero).

Reply:
Right. Thinking more about it, I think I am going to use `-1` to mean that all new paths are completely disabled, so that we don't call `prctl` by default, and do not print the slack values at all. With `0`, we rely on system default, and then print out the actual values. With `> 0`, we both override the value and print the current slack values.

This insulates current code from this experimental addition better.